module/validation/receipt_validator_test.go

```diff
@@ -247,7 +248,7 @@ func (s *ReceiptValidationSuite) TestReceiptInvalidResultChain() {
 		mock.Anything,
 		mock.Anything).Return(true, nil).Maybe()
 
-	err := s.receiptValidator.Validate(one(receipt))
+	err := s.receiptValidator.Validate(receipt)
 	s.Require().Error(err, "should reject invalid previous result")
```

Comment:
⚠️
Same here: I think your are erroring on a different failure path than the one you intended to test.

```suggestion
	s.Require().Error(err, "should reject invalid previous result")
	s.Assert().True(engine.IsInvalidInputError(err), err)
```